src/jobs_auto_report/main.py

```diff
@@ -0,0 +1,40 @@
+from opensearchpy import OpenSearch
+from job_assets import Assets
+from slack_sdk.webhook import WebhookClient
+from action import Action
+from datetime import datetime, timedelta
+import os
+
+
+os_usr = os.environ['OPENSEARCH_USERNAME']
+os_pwd = os.environ['OPENSEARCH_PASSWORD']
+os_host = os.environ['OPENSEARCH_HOST']
+
+slack_webhook_url = os.environ['SLACK_WEBHOOK_URL']
+
+client = OpenSearch(
+    os_host,
+    http_auth=(os_usr, os_pwd)
+)
+
+index_name = 'jobs-*'
+
+data = {}
+
+now = datetime.now()
+a_week_ago = now - timedelta(weeks=1)
+
+action = Action(client, index_name)
+
+success_rate = action.get_success_rate(a_week_ago, now)
+data['success_rate'] = success_rate
+
+number_of_jobs_triggered = action.get_number_of_jobs_triggered(a_week_ago, now)
+data['jobs_triggered'] = number_of_jobs_triggered
+
+average_job_duration = action.get_average_duration_of_jobs(a_week_ago, now)
+data['average_job_duration'] = average_job_duration
+
+url = os.environ['SLACK_WEBHOOK_URL']
+webhook = WebhookClient(url)
+response = webhook.send(text=f"Weekly Report:\n```{data}```")
```

Comment:
Please change your IDE configuration to end files with a blank line (meaning you shouldn't see this icon at end of files in github :no_entry:  )